src/main/java/generator/ModelBuilder.java

```diff
@@ -232,14 +238,26 @@ public ModelBuilder addPedestrianCrossing(int peopleCount) {
         pedestrianCrossing.addDistance(distance);
         pedestrianCrossing = fillDataProps(pedestrianCrossing, distance, "pedestrian_crossing");
 
-        for (Map.Entry<Integer, Lane> lane : model.getLanes().get(Model.Side.LEFT).entrySet())
+        for (Map.Entry<Integer, Lane> lane : model.getLanes().get(Model.Side.LEFT).entrySet()) {
+            Lane l = lane.getValue();
+            l.addLane_has_object(pedestrianCrossing);
+            pedestrianCrossing.addIs_on_lane(l);
             model.getObjects().get(lane.getValue()).add(pedestrianCrossing);
+        }
 
-        for (Map.Entry<Integer, Lane> lane : model.getLanes().get(Model.Side.CENTER).entrySet())
+        for (Map.Entry<Integer, Lane> lane : model.getLanes().get(Model.Side.CENTER).entrySet()) {
+            Lane l = lane.getValue();
+            l.addLane_has_object(pedestrianCrossing);
+            pedestrianCrossing.addIs_on_lane(l);
             model.getObjects().get(lane.getValue()).add(pedestrianCrossing);
+        }
 
-        for (Map.Entry<Integer, Lane> lane : model.getLanes().get(Model.Side.RIGHT).entrySet())
+        for (Map.Entry<Integer, Lane> lane : model.getLanes().get(Model.Side.RIGHT).entrySet()) {
+            Lane l = lane.getValue();
+            l.addLane_has_object(pedestrianCrossing);
+            pedestrianCrossing.addIs_on_lane(l);
             model.getObjects().get(lane.getValue()).add(pedestrianCrossing);
+        }
```

Comment:
Dokładnie taki sam kod, usuń powtórzenie, jak w kodzie klikasz ctrl+c - ctrl+v na więcej niż 1 linii, to niech Ci się zapali czerwona lampka, że masz powtórzenie kodu ;d

Reply:
A to z grupy generatora jest